docs/guide/src/example-plugin/plugin.go

```diff
@@ -6,55 +6,115 @@ import (
 	"github.com/tendermint/go-wire"
 )
 
+//-----------------------------------------
+// Structs
+//   * Note the fields in each struct may be expanded/modified
+
+// Plugin State Struct
+//   * Intended to store the current state of the plugin
+//     * This example contains a field which holds the execution count
+//   * Used by go-wire as the encoding/decoding struct to hold the plugin state
+//   * All fields must be exposed (for go-wire)
```

Comment:
for go-wire, or any other external serialization lib (you don't have to use go-wire for the state in your basecoin plugin)